dlt/common/runtime/exec_info.py

```diff
@@ -62,11 +62,23 @@ def is_github_actions() -> bool:
 
 def is_notebook() -> bool:
     try:
-        return bool(str(get_ipython()))  # type: ignore
+        # Import get_ipython from IPython
+        from IPython import get_ipython
+        # Get the class name of the current IPython instance
+        shell = get_ipython().__class__.__name__
```

Comment:
did you have a false positive? so notebook detected while not being in the environment? what was wrong with previous function?

Reply:
so you can have `get_ipython()` returning true without actually being a notebook?